WordPress/gutenberg
# Allow drag and drop to create Rows and Galleries

## What?


Implements #13202

Enhances drag and drop so that, when hovering over or around blocks in a vertical layout, dropzones appear around the sides to allow for grouping blocks horizontally into Rows. 

Additionally, if an Image block is dropped beside another Image block, they will transform into a Gallery. 

Is this a good addition? Feedback appreciated!


## Testing Instructions





1. Open a post and multiple blocks;
2. Try dragging them around each other to see the lateral dropzones appear;
3. Try dropping a block into a lateral dropzone and check that a Row is created with both blocks;
4. Drag an image from outside into the editor and check that it can also be formed into  a Row with existing blocks;
5. Drag an Image block beside another Image block and check that they become a Gallery block;
6. Try to break this in imaginative ways.

Todo: 
* make images dragged in from outside onto Image blocks become Galleries too (currently they just become Rows)
* address test failures



https://github.com/WordPress/gutenberg/assets/8096000/bcc04337-5e0e-437c-b4af-d3cab022e7b2
